spec/lib/payment_processor/go_cardless/transaction_spec.rb

```diff
@@ -58,18 +59,67 @@ module GoCardless
-              params: {
-                amount: amount_in_euros * 100,
-                currency: 'EUR',
-                links: { mandate: 'MA00000' },
-                metadata: { customer_id: 'CU00000' }
+          context 'with EUR' do
+            it 'creates a transaction with the right params and charge date' do
+              expect_any_instance_of(
+                  GoCardlessPro::Services::PaymentsService
+              ).to receive(:create).with(
+                       params: {
+                           amount: amount_in_euros * 100,
+                           currency: 'EUR',
+                           links: { mandate: 'MA00000' },
+                           metadata: { customer_id: 'CU00000' },
+                           charge_date: mandate.next_possible_charge_date
+                       }
+                   )
+              subject
+            end
+          end
+
+          context 'with GBP' do
+
+            let(:amount_in_gbp) { 11.11 }
+            let(:completed_gbp_flow) do
+              instance_double('GoCardlessPro::Resources::RedirectFlow',
+                              links: double(customer: 'CU00000', mandate: 'MA9999')
+              )
+            end
+            let(:gbp_mandate) do
+              instance_double('GoCardlessPro::Resources::Mandate',
+                              id: 'MA9999', scheme: 'bacs', next_possible_charge_date: '2016-05-22'
+              )
+            end
+            let(:gbp_options) do
+              {
+                  amount: amount_in_gbp,
+                  currency: 'GBP',
+                  user: { email: "bob@example.com", name: 'Bob' },
+                  page_id: page_id,
+                  redirect_flow_id: 'RE00000',
+                  session_token: "4f592f2a-2bc2-4028-8a8c-19b222e2faa7"
               }
-            )
-            subject
+            end
+
+            it 'creates a transaction with the right params and charge date' do
+              allow_any_instance_of(GoCardlessPro::Services::RedirectFlowsService).to receive(:complete).and_return(completed_gbp_flow)
+              allow_any_instance_of(GoCardlessPro::Services::RedirectFlowsService).to receive(:get).and_return(completed_gbp_flow)
+              allow_any_instance_of(GoCardlessPro::Services::MandatesService).to receive(:get).and_return(gbp_mandate)
+              allow(PaymentProcessor::Currency).to receive(:convert).and_return(double(cents: amount_in_gbp*100))
+
+              expect_any_instance_of(
+                  GoCardlessPro::Services::PaymentsService
+              ).to receive(:create).with(
+                       params: {
+                           amount: amount_in_gbp * 100,
+                           currency: 'GBP',
+                           links: { mandate: 'MA9999' },
+                           metadata: { customer_id: 'CU00000' },
+                           charge_date: "2016-06-#{Settings.gocardless.gbp_charge_day}"
```

Comment:
this test works for me. Only thing I would add is that I would do a version where change the Setting in a before block (just `Settings.gocardless.gbp_charge_day = ''`) and then test that `charge_date` is `mandate.next_possible_charge_date`